src/promnesia/sources/viber_desktop.py

```diff
@@ -0,0 +1,164 @@
+"""
+Adapted from `telegram.py` to read from `~/.ViberPC/XYZ123/viber.db`
+"""
+
+import json
+import logging
+import textwrap
+from pathlib import Path
+from typing import Optional, TypeVar, Union
+from urllib.parse import \
+    unquote  # TODO mm, make it easier to rememember to use...
+
+from ..common import (Loc, PathIsh, Results, Visit, echain, extract_urls,
+                      from_epoch, get_logger)
+
+# TODO potentially, belongs to my. package
```

Comment:
Ideally this module (i.e. the query/parsing part) would be part of HPI (so viber data can be used by other apps).
The only reason it's not is that the promnesia module existed before I started HPI project :)
But totally ok to keep here for now, can always refactor later if needed.

Reply:
(read it before my own suggestion on that, [below](https://github.com/karlicoss/promnesia/pull/204#pullrequestreview-597263776)).
Yes i agree.
Will transplant part of this PR in HPI.

I will wait for your review to finish here, before spreading stuff in 2 projects.